src/mainwindow.py

```diff
@@ -3,170 +3,315 @@
+                    cat = choice(data[letter])
+                    # update the cat data dict
+                    cat_data.update({"name": cat["name"]})
+                    cat_data.update({"gender": cat["gender"]})
+
+                # get 5 random hobbies
+                async with self.session.get(
+                    "https://gist.githubusercontent.com/mbejda/" +
+                    "453fdb77ef8d4d3b3a67/raw/e8334f09109dc212892406e25fdee03efdc23f56/" +
+                    "hobbies.txt"
+                ) as res:
+                    text = await res.text()
+                    # split the raw text of hobbies into a list
+                    all_hobbies = text.split("\n")
+                    # get 5 of those hobbies
+                    hobby_list = sample(all_hobbies, 5)
+                    # join those 5 hobbies into a bulleted list (string)
+                    list_of_hobbies = "\n •".join(hobby_list)
+                    hobbies = f"Hobbies:\n •{list_of_hobbies}"
+                    # update the cat_data dict
+                    cat_data.update({"hobbies": hobbies})
+
+                # get a random age between 1 and 15 (avg lifespan of a cat)
+                age = str(randint(1, 15))
+                # update the cat data dict
+                cat_data.update({"age": age})
+
+                # get a random number of miles away between 1 and 5
+                miles = randint(1, 5)
+                location = f"{miles} miles away"
+                # update the cat data dict
+                cat_data.update({"location": location})
+            self.cats.append(cat_data)
 
     def all_children(self):
+        '''Used to get all children of the root window
+
+        Returns
+        -------
+            A `list` of tkinter objects that are children of the root window'''
+
+        # get children
         children = self.root.winfo_children()
+        # loop through the children
         for item in children:
+            # if the child has children, add them to the list
             if item.winfo_children():
                 children.extend(item.winfo_children())
+        # return the full list of children
         return children
 
+    def new_image(self, cat=None):
+        '''Main functionality of the application
 
-    def new_image(self, cat= None):
+        param
+            cat: dict -- if you want to run the function without getting a new cat'''
+
+        # if the previous image was a jumpscare, resize the window and reset the variable
         if self.jumpscare:
             self.root.maxsize(400, 500)
             self.jumpscare = False
+
+        # get all children of the root window
         widget_list = self.all_children()
         for item in widget_list:
+            # forget packs all of the children (This clears the window)
             item.pack_forget()
-        self.frame = tk.Frame(self.root, bg = "black")
+
+        # make a new Frame
+        self.frame = tk.Frame(self.root, bg="black")
```

Comment:
Define self.frame in __init__()

Reply:
I forget all of the packs every time self.new_image() is run, so the frame would be deleted if it was just made in init.